src/pages/forum/Forum.tsx

```diff
@@ -1,37 +1,71 @@
 import React, { FC, ReactElement, useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import { Link } from 'react-router-dom';
 
+import { getTopics, createTopic } from 'api/forumApi';
 import { BUTTON_TEXTS } from 'constants/buttonConstants';
 import { PAGE_NAMES } from 'constants/commonConstants';
+import { ERROR_CONSTANTS } from 'constants/errorConstants';
 import { LINK_TEXTS } from 'constants/linkConstants';
 import { FORUM_CONSTANTS } from 'constants/forumConstants';
 import { PAGE_SIZE } from 'constants/paginationConstants';
 import { ROUTE_CONSTANTS } from 'constants/routeConstants';
-import { Button, PageMeta, Pagination } from 'components';
+import { Button, Input, Modal, PageMeta, Pagination } from 'components';
+import { setUserPending } from 'store/user/actions';
+import { userUserDataSelector } from 'store/user/selectors';
 import { formatBigNumbers, formatDate } from 'utils';
-import { topicsInfo, ITopic } from './mock';
 
 import './forum.scss';
 
+export interface ITopic {
+  id: number;
+  title: string,
+  messagesCount: number,
+  createdAt: number,
+}
+
 type TopicState = ITopic[] | []
```

Comment:
❗️
Нет особого смысл в этом, `ITopic[]` вполне себе подразумевает, что он может быть пустым)
```suggestion
type TopicState = ITopic[]
```